### PR TITLE
[Doc] PR for Document LDAP support for Platform ( CLI / API ) - Part 2

### PR DESCRIPTION
**Pull request by lizayugabyte**
Addresses https://yugabyte.atlassian.net/browse/PLAT-2730

Relates to https://github.com/yugabyte/yugabyte-db/pull/11023

Preview: https://deploy-preview-11111--infallible-bardeen-164bc9.netlify.app/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication/

### CODE REVIEW

**Comment by lizayugabyte** [comment]
@sridhar-yb, kc-yb, Arnav15, could you, please, review the doc so I would be able to merge the PR in time for the release

**Comment by lizayugabyte** [comment]
@TaranSinghania-yb With 5 reviewers on the list, I don't think we'd be able to finish the doc in time for 2.12 release. If the file changes delivered by this PR are fine, I'm going to merge. Once the UI is available, I'll update the doc and create a new PR.

**Comment by TaranSinghania-yb** [comment]
Hi @lizayugabyte, we just need to wait for @Arpit-yb's review. We need to add some Windows AD specific information here.

**Review by Arpit-yb — CHANGES_REQUESTED** [review]
Can we add something for windows AD like setup, something like - "The user should have permissions to query its attributes/properties from the LDAP server otherwise the Platform will default its role as ReadOnly"

File: docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md [hunk 1]
```diff
@@ -88,4 +101,7 @@ For additional information, see [Update a configuration key](https://yugabyte.st
 
 ## Define the Yugabyte Platform Role
 
-You need to define a Yugabyte Platform-specific role for each user on your LDAP server by setting the a `YugabytePlatformRole` annotation.
+You need to define a Yugabyte Platform-specific role for each user on your LDAP server by setting the `YugabytePlatformRole` annotation. The value set for this annotation is read during the Yugabyte Platform login. Note that this value cannot be changed from the Yugabyte Platform side; if the value is modified on the LDAP server, the change is propagated to Yugabyte Platform and automatically updated during login. Password updates are also automatically handled.
```
[the remark this answers is not in the document]
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:104]: Deleted this sentence.

**Comment by lizayugabyte** [comment]
@TaranSinghania-yb @Arpit-yb Could you take a look at the latest changes and let me know if this PR is good to go

File: docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md [hunk 2]
```diff
@@ -88,4 +101,8 @@ For additional information, see [Update a configuration key](https://yugabyte.st
 
 ## Define the Yugabyte Platform Role
 
-You need to define a Yugabyte Platform-specific role for each user on your LDAP server by setting the a `YugabytePlatformRole` annotation.
+You need to define a Yugabyte Platform-specific role for each user on your LDAP server by setting the `YugabytePlatformRole` annotation. The value set for this annotation is read during the Yugabyte Platform login. Note that if the value is modified on the LDAP server, the change is propagated to Yugabyte Platform and automatically updated during login. Password updates are also automatically handled.
+
+If the role is not specified, users are created with ReadOnly privileges by default, which can be modified by the local super admin.
+
+When LDAP is set up on a Windows Active Directory (AD) server, the user is expected to have permissions to query the AD's properties from that server. If the permissions have not been granted, Yugabyte Platform defaults its role to ReadOnly, which can later be modified by the local super admin.
```
Comment (Arpit-yb) [inline docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:108]: "the user is expected to query its properties"
Replace "AD's properties" -> "user's properties"

Rest everything looks good

**Review by Arpit-yb — COMMENTED** [review]
PR looks OK to me, @lizayugabyte just modify this minor line and we should be good to merge.
Will wait for @TaranSinghania-yb's approval

File: docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md (hunk 2 above)
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:108]: changed AD's to user's

**Comment by lizayugabyte** [comment]
@TaranSinghania-yb Could you approve the PR, please?

**Review by TaranSinghania-yb — CHANGES_REQUESTED** [review]

File: docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md [hunk 3]
```diff
@@ -56,7 +57,7 @@ To enable LDAP authentication for Yugabyte Platform login, you need to use a num
   --data-raw '389'
   ```
 
-- The base distinguished name (DN) to enable restriction of users and user groups, as follows:
+- The base distinguished name (DN) `yb.security.ldap.ldap_basedn` to enable restriction of users and user groups, as follows:
```
[the remark this answers is not in the document]
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:60]: Added this example as:

Replace `[LDAP DN]` with the actual value, as per the following example: 
`,DC=yugabyte,DC=com`

File: docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md [hunk 4]
```diff
@@ -68,7 +69,19 @@ To enable LDAP authentication for Yugabyte Platform login, you need to use a num
...
+
+  ```shell
+  curl --location --request PUT 'https://10.9.140.199/api/v1/customers/f3a63f07-e3d6-4475-96e4-57a6453072e1/runtime_config/00000000-0000-0000-0000-000000000000/key/yb.security.ldap.ldap_dn_prefix' \
+  --header 'X-AUTH-YW-API-TOKEN: 5182724b-1891-4cde-bcd1-b8f7a3b7331e' \
+  --header 'Content-Type: text/plain' \
+  --header 'Cookie: csrfCookie=d5cdb2b36b00fcad1f4fdb24605aee412f8dfaa0-1641544510767-641be933bf684abcade3c592' \
+  --data-raw '[LDAP DN PREFIX]'
+  ```
+
+  <br>Replace `[LDAP DN PREFIX]` with the actual value.
```
[the remark this answers is not in the document]
Comment (TaranSinghania-yb) [reply docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:82]: Basically:` LDAP DN Prefix + user name (entered in login form) + Base DN` - concatenation

[the remark this answers is not in the document]
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/administer-yugabyte-platform/ldap-authentication.md:82]: Added these two points

**Review by TaranSinghania-yb — APPROVED** [review]
LGTM for API only. Thanks so much!

**Review by Arpit-yb — APPROVED** [review]
LGTM, Nice work @lizayugabyte!

**Pull request merged**